DentOS_Framework/DentOsTestbed/src/dent_os_testbed/utils/test_utils/cleanup_utils.py

```diff
@@ -0,0 +1,108 @@
+from dent_os_testbed.constants import DEFAULT_LOGGER
+from dent_os_testbed.lib.ip.ip_address import IpAddress
+from dent_os_testbed.lib.ip.ip_link import IpLink
+from dent_os_testbed.lib.ip.ip_route import IpRoute
+from dent_os_testbed.lib.tc.tc_qdisc import TcQdisc
+from dent_os_testbed.logger.Logger import AppLogger
+from dent_os_testbed.utils.test_utils.tb_utils import (
+    tb_get_all_devices
+)
+from dent_os_testbed.utils.test_utils.tgen_utils import (
+    tgen_utils_get_dent_devices_with_tgen
+)
+
+
+async def post_test_cleanup_qdiscs(testbed):
+    """Removes all non-default qdiscs created during test. 
+    Can be used separately or by using `cleanup_qdiscs` fixture."""
+    logger = AppLogger(DEFAULT_LOGGER)
+    dev = await tb_get_all_devices(testbed)
+    dev = dev[0]
+    logger.info("Clearing TC")
+    out = await TcQdisc.show(
+        input_data=[{dev.host_name: [{"options": "-j"}]}], 
+        parse_output=True
+    )
+    qdiscs_info = out[0][dev.host_name]["parsed_output"]
+    for qdisc_obj in qdiscs_info:
+        if qdisc_obj["kind"] != "noqueue":
+            await TcQdisc.delete(
+                input_data=[{dev.host_name: [{"dev": qdisc_obj["dev"], "direction": qdisc_obj["kind"]}]}]
+            )
+
+
+async def post_test_cleanup_bridges(testbed):
+    """Removes all bridges created during test. 
+    Can be used separately or by using `cleanup_bridges` fixture."""
+    logger = AppLogger(DEFAULT_LOGGER)
+    dev = await tb_get_all_devices(testbed)
+    dev = dev[0]
+    logger.info("Clearing bridges")
+    out = await IpLink.show(
+        input_data=[{dev.host_name: [{"link_type": "bridge", "cmd_options": "-j"}]}], 
+        parse_output=True
+    )
+    bridges_info = out[0][dev.host_name]["parsed_output"]
+    if bridges_info:
+        await IpLink.delete(input_data=[{dev.host_name: [
+            {"device": bridge_obj["ifname"]} for bridge_obj in bridges_info]}
+        ])
+
+
+async def post_test_cleanup_vrfs(testbed):
+    """Removes all VRFs created during test. 
+    Can be used separately or by using `cleanup_vrfs` fixture."""
+    logger = AppLogger(DEFAULT_LOGGER)
+    dev = await tb_get_all_devices(testbed)
+    dev = dev[0]
+    logger.info("Deleting VRFs")
+    out = await IpLink.show(
+        input_data=[{dev.host_name: [{"link_type": "vrf",  "cmd_options": "-j"}]}], 
+        parse_output=True
+    )
+    vrfs_info = out[0][dev.host_name]["parsed_output"]
+    if all(vrfs_info):
+        await IpLink.delete(input_data=[{dev.host_name: [
+            {"device": vrf_obj["ifname"]} for vrf_obj in vrfs_info
+        ]}])
+
+
+async def post_test_cleanup_ip_addrs(testbed):
+    """Removes all IP addresses configured during test. 
+    Can be used separately or by using `cleanup_addrs` fixture."""
+    logger = AppLogger(DEFAULT_LOGGER)
+    logger.info("Deleting IP addresses from interfaces")
+    tgen_dev, dent_dev = await tgen_utils_get_dent_devices_with_tgen(testbed, [], 4)
```

Comment:
The assumption that we flush the ip addresses only on port connected to TG right?